Source/Core/Core/PowerPC/JitArm64/JitArm64_RegCache.h

```diff
@@ -180,6 +180,57 @@ class Arm64RegCache
   // Requires unlocking after done
   Arm64Gen::ARM64Reg GetReg();
 
+  class ScopedARM64Reg
+  {
+    Arm64Gen::ARM64Reg m_reg = Arm64Gen::ARM64Reg::INVALID_REG;
+    Arm64RegCache* m_gpr = nullptr;
+
+    inline void Invalidate()
+    {
+      m_reg = Arm64Gen::ARM64Reg::INVALID_REG;
+      m_gpr = nullptr;
+    }
```

Comment:
Private functions and variables should be after the public ones, and there should be a `private:` line before them.